lib/valkyrie/persistence/postgres/query_service.rb

```diff
@@ -35,6 +35,13 @@ def find_all_of_model(model:)
       end
     end
 
+    # Count all records for a specific resource type
+    # @param [Class] model
+    # @return integer
+    def count_all_of_model(model:)
+      orm_class.where(internal_resource: model.to_s).count
```

Comment:
Can we use `orm_class.count` instead of retrieving all of the objects to count them? That should make this scale and perform much better.

Reply:
This concern applies to the other implementations as well.

It seems reasonable to support some of these naively if others have efficient implementations, but if all are just loading up objects and counting them i wonder if it isn't better to avoid adding the API footprint of the new query.

Reply:
@no-reply I think Solr and Postgres (and anything else that has a query API) should be able to do this task much more efficiently than loading-and-counting.  But if that isn't the case, then I agree it would probably not make any sense to add queries for it.

Somewhat related: @awead has mentioned and shown in our workshop some helpers that add ActiveRecord-style `all`, `count`, `last`, etc. to models.  I wonder if there is some other pattern we want to explore for sharing convenience methods where there isn't a performance angle, but just making the API a little easier to work with?

Reply:
I've done a little digging, and it looks like `.count` transforms the ActiveRecord relation to a `COUNT` query, so I now think this is fine.